### PR TITLE
Allow the autopilot plugin to deploy new applications

### PR DESCRIPTION
**Pull request by pivotal-topher-bullock**
Fixes Issue #14 

[Finishes #108651710]

Signed-off-by: Dave Protasowski <dprotaso@gmail.com>

### CODE REVIEW

**Comment by pivotal-topher-bullock** [comment]
Added an additional fix for #4
Blindly attempts to delete the app so the rename back to the original app name succeeds.
:dancers:

File: autopilot.go [hunk 1]
```diff
@@ -23,39 +25,69 @@ func main() {
...
+		},
+		// push
+		{
+			Forward: func() error {
+				return appRepo.PushApplication(manifestPath, appPath)
+			},
+			ReversePrevious: func() error {
+				// If the app cannot start we'll have a lingering application
+				// We delete this application so that the rename can succeed
+				appRepo.DeleteApplication(appName)
```
Comment (xoebus) [inline autopilot.go]: If this fails we should probably return the error so that things stop and the user can fix it manually.

**Comment by xoebus** [comment]
1 thing I left a comment on but LGTM.

File: autopilot.go (hunk 1 above)
Comment (pivotal-topher-bullock) [inline autopilot.go]: @xoebus
@dprotaso and I debated returning the error from DeleteApplication if any occurs, but there's no guarantee that when the push failed the app was created at all. In that case we'd be returning a lot of failures to delete the non-existent app, and not continue on to the rename.

If for some reason we couldn't delete the application (but it existed) then the following rename attempt would error out.

**Comment by pivotal-topher-bullock** [comment]
See: https://github.com/concourse/autopilot/pull/19/files#r54133894

File: autopilot.go (hunk 1 above)
Comment (xoebus) [inline autopilot.go]: Good thinking. I'll try to merge this today.

**Comment by dprotaso** [comment]
@xoebus Pinged you on slack (not sure if you saw it) do you mind cutting a github release after the merge? So our pipelines will pick up the new changes.

thanks

**Comment by xoebus** [comment]
Thanks!

**Pull request merged**